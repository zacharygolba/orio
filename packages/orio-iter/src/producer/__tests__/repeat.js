// @flow

import Repeat from '../repeat'

const ITEM = 'test'
let producer

beforeEach(() => {
  producer = new Repeat(ITEM)
})

test('#constructor()', () => {
  expect(producer).toMatchSnapshot()
})

test('#drop()', () => {
  expect(producer.drop()).toBeUndefined()
  expect(producer.next()).toMatchSnapshot()
  expect(producer).toMatchSnapshot()
})
