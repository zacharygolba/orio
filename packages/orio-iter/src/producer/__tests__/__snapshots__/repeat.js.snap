// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#constructor() 1`] = `
Repeat {
  "done": false,
  "value": "test",
}
`;

exports[`#drop() 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#drop() 2`] = `
Repeat {
  "done": true,
  "value": "test",
}
`;
